Include/cpython/pyatomic_msc.h

```diff
@@ -0,0 +1,860 @@
+#ifndef Py_ATOMIC_MSC_H
+#  error "this header file must not be included directly"
+#endif
+
+// This is the implementation of Python atomic operations for MSVC if the
+// compiler does not support C11 or C++11 atomics.
+
+#include <intrin.h>
+
+
+static inline int
+_Py_atomic_add_int(int *address, int value)
+{
+    return (int)_InterlockedExchangeAdd((volatile long*)address, (long)value);
+}
+
+static inline int8_t
+_Py_atomic_add_int8(int8_t *address, int8_t value)
+{
+    return (int8_t)_InterlockedExchangeAdd8((volatile char*)address, (char)value);
+}
+
+static inline int16_t
+_Py_atomic_add_int16(int16_t *address, int16_t value)
+{
+    return (int16_t)_InterlockedExchangeAdd16((volatile short*)address, (short)value);
+}
+
+static inline int32_t
+_Py_atomic_add_int32(int32_t *address, int32_t value)
+{
+    return (int32_t)_InterlockedExchangeAdd((volatile long*)address, (long)value);
+}
+
+static inline int64_t
+_Py_atomic_add_int64(int64_t *address, int64_t value)
+{
+#if defined(_M_X64) || defined(_M_ARM64)
+    return (int64_t)_InterlockedExchangeAdd64((volatile __int64*)address, (__int64)value);
+#else
+    for (;;) {
```

Comment:
pycore_atomic.h uses a different style which may be more readable:

``do { ... } while (CMPXHG(...) == old_value);``

Reply:
I'm fine with the current style (1) ``for (;;)``.